client/dashboard/sites/settings-wp-version/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { DataForm } from '@automattic/dataviews';
+import { useQuery, useMutation } from '@tanstack/react-query';
+import { notFound } from '@tanstack/react-router';
+import {
+	Card,
+	CardBody,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Button,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import {
+	siteQuery,
+	siteWordPressVersionQuery,
+	siteWordPressVersionMutation,
+} from '../../app/queries';
+import PageLayout from '../../components/page-layout';
+import { getWordPressVersionTagName } from '../../utils/wp-version';
+import SettingsPageHeader from '../settings-page-header';
+import { canUpdateWordPressVersion } from './utils';
+import type { Field, SimpleFormField } from '@automattic/dataviews';
+
+const fields: Field< { version: string } >[] = [
+	{
+		id: 'version',
+		Edit: 'select',
+		elements: [
+			{ value: 'latest', label: getWordPressVersionTagName( 'latest' ) },
+			{ value: 'beta', label: getWordPressVersionTagName( 'beta' ) },
+		],
+	},
+];
+
+const form = {
+	type: 'regular' as const,
+	fields: [ { id: 'version', labelPosition: 'none' } as SimpleFormField ],
+};
+
+export default function WordPressVersionSettings( { siteSlug }: { siteSlug: string } ) {
+	const { data: site } = useQuery( siteQuery( siteSlug ) );
```

Comment:
Should this be a Modal instead of content of a new route? I think the designs don't show the arrow for this one..

Reply:
Hmm, I see the arrow here: QOBjujZah0UlGDDdLKZhzi-fi-1_48466

I explicitly asked @matt-west about individual routes here: QOBjujZah0UlGDDdLKZhzi-fi-1_48466#1244125179

Reply:
Okay. I asked because I checked the design below and doesn't have the arrow 😄

Reply:
The arrow comes from the `SummaryButton` component itself though... maybe it should be updated to not show it when it is `disabled`. 🤔